### PR TITLE
Implement more `Bitcast` conversions.

### PR DESCRIPTION
**Pull request by sunfishcode**
Implement `Bitcast` conversions between more combinations of f32, f64, pointer, and length types.

### CODE REVIEW

**Review by alexcrichton — COMMENTED** [review]
Thanks! From Zulip you mention:

> I'm thinking about whether to remove the Bitcast enum and just have each backend match on the (to, from) pair itself.

Originally I added `Bitcast` since a good number of pairs of bitcasts weren't ever needed and it felt nice to ensure that code generators wouldn't have to have dead code to implement them. I agree though that as this set is growing it's becoming more unwieldy so that may no longer be the case and it might be worth just having a `from: Type, to: Type` structure.

File: crates/core/src/abi.rs [hunk 1]
```diff
@@ -1910,11 +1922,22 @@ fn cast(from: WasmType, to: WasmType) -> Bitcast {
...
         (Pointer, I32) => Bitcast::PToI32,
+        (F32, Pointer) => Bitcast::F32ToP,
+        (Pointer, F32) => Bitcast::PToF32,
+
+        (I32, Length) => Bitcast::I32ToL,
+        (Length, I32) => Bitcast::LToI32,
+        (I64, Length) => Bitcast::I64ToL,
+        (Length, I64) => Bitcast::LToI64,
+        (F32, Length) => Bitcast::F32ToL,
+        (Length, F32) => Bitcast::LToF32,
```
[the remark this answers is not in the document]
Comment (sunfishcode) [reply crates/core/src/abi.rs:1940]: They do join to a `Length`. When they do, we need to be able to convert `f32` values into and out of the `Length`.

[the remark this answers is not in the document]
Comment (alexcrichton) [reply crates/core/src/abi.rs:1940]: Oh right, I don't know why I keep forgetting that...

**Comment by sunfishcode** [comment]
To reduce the level of boilerplate, I've now added a `Sequence` bitcast, which can pair together two bitcasts, so that we can translate "anything to PointerOrI64" to the two-step conversion "anything to I64, I64 to PointerOrI64", which factors out a lot of the redundancy.

**Review by alexcrichton — APPROVED** [review]
Nice I like how this turned out 👍

**Pull request merged**